lib/src/screens/new_wallet/new_wallet_page.dart

```diff
@@ -105,18 +105,50 @@ class _WalletNameFormState extends State<WalletNameForm> {
               padding: EdgeInsets.only(top: 24),
               child: Form(
                 key: _formKey,
-                child: TextFormField(
-                  onChanged: (value) => _walletNewVM.name = value,
-                  controller: _controller,
-                  textAlign: TextAlign.center,
-                  style: TextStyle(
-                      fontSize: 20.0,
-                      fontWeight: FontWeight.w600,
-                      color: Theme.of(context).primaryTextTheme.title.color),
-                  decoration: InputDecoration(
-                    suffixIcon: IconButton(
+                child: Stack(
+                  alignment: Alignment.centerRight,
+                  children: [
+                    TextFormField(
+                      onChanged: (value) => _walletNewVM.name = value,
+                      controller: _controller,
+                      textAlign: TextAlign.center,
+                      style: TextStyle(
+                          fontSize: 20.0,
+                          fontWeight: FontWeight.w600,
+                          color:
+                              Theme.of(context).primaryTextTheme.title.color),
+                      decoration: InputDecoration(
+                        hintStyle: TextStyle(
+                            fontSize: 18.0,
+                            fontWeight: FontWeight.w500,
+                            color: Theme.of(context)
+                                .accentTextTheme
+                                .display3
+                                .color),
+                        hintText: S.of(context).wallet_name,
+                        focusedBorder: UnderlineInputBorder(
+                            borderSide: BorderSide(
+                                color: Theme.of(context)
+                                    .accentTextTheme
+                                    .display3
+                                    .decorationColor,
+                                width: 1.0)),
+                        enabledBorder: UnderlineInputBorder(
+                          borderSide: BorderSide(
+                              color: Theme.of(context)
+                                  .accentTextTheme
+                                  .display3
+                                  .decorationColor,
+                              width: 1.0),
+                        ),
+                      ),
+                      validator: WalletNameValidator(),
+                    ),
+                    IconButton(
                       onPressed: () async {
                         final rName = await generateName();
+                        FocusManager.instance.primaryFocus?.unfocus();
```

Comment:
is necessary that here ?

Reply:
@mkyq , I have tested this again by commenting this statement ` FocusManager.instance.primaryFocus?.unfocus();` and this is essential. Because without this the keyboard won't dismiss if the user decides to switch from typing to pressing the icon button.